libc/src/gpu/rpc_recv.cpp

```diff
@@ -17,9 +17,12 @@ namespace LIBC_NAMESPACE {
 static_assert(sizeof(rpc_port_t) == sizeof(rpc::Client::Port), "ABI mismatch");
 static_assert(sizeof(rpc_buffer_t) == sizeof(rpc::Buffer), "ABI mismatch");
 
-LLVM_LIBC_FUNCTION(void, rpc_recv, (rpc_port_t * handle)) {
+LLVM_LIBC_FUNCTION(void, rpc_recv,
+                   (rpc_port_t * handle, rpc_callback_t callback, void *data)) {
```

Comment:
The `rpc_port_t` is more or less a type-erased buffer. We convert it back to the real type since we know what it is internally. To the user, it's just like 32 bytes.